Automattic/jetpack
# Jetpack AI: open upgrade path on new tab, part 1

Part of #37458.

This change will make the upgrade button from the usage panel opens on a new tab instead of redirecting the editor tab.

## Proposed changes:

* Change target of buttons and links to `_blank`
* Remove calls to `autoSaveAndRedirect` hook

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* run this branch on a fresh JN site
* connect Jetpack then go to the block editor
* enable debug messages by running `localStorage.setItem( 'debug', '*' );` on the console
* look for the AI Assistant section on the "Post" tab of the Settings sidebar
* look for the "Available Requests" section:

<img width="300" alt="Screenshot 2024-05-20 at 17 51 48" src="https://github.com/Automattic/jetpack/assets/6760046/cf9fc711-f86a-4d7a-9764-71dc740977fb">

* click the "Upgrade" button
* confirm it opens the upgrade flow on a new tab
* on the editor tab, confirm the tracking event was fired as expected:

<img width="350" alt="Screenshot 2024-05-20 at 17 53 14" src="https://github.com/Automattic/jetpack/assets/6760046/7175934d-0e25-4023-a638-259d7bd7168c">

* proceed with the upgrade flow on the new tab, buy the plan using credits
* go back to the editor tab and refresh the page to get the new plan info
* follow the same procedure to buy the other tiers, confirming that
   * it opens on a new tab
   * the tracking event is fired as before
* follow the same procedure on the last tier, when the button changes to "Contact Us", confirming that
   * it opens on a new tab
   * the tracking event is fired as before
